gdpr_assist/anonymiser.py

```diff
@@ -208,21 +209,44 @@ def anonymise_field(instance, field_name):
     if cls._meta.pk.name == field_name:
         raise AnonymiseError('Cannot anonymise primary key')
 
-    # Find field
-    field = cls._meta.get_field(field_name)
-    value = getattr(instance, field_name)
+    privacy_meta = getattr(instance, GDPR_PRIVACY_INSTANCE_NAME)
+
+    if field_name in privacy_meta.fk_fields:
+        field = getattr(instance, field_name)
+
+        if field is not None:
+            instance._log_gdpr_recursive(user=user, start=True)
+            field.anonymise(user=user)
+            instance._log_gdpr_recursive(user=user, start=False)
```

Comment:
Could you explain what is the flow here to log things? Could we use a context manager instead? Something like:

```python
with log_gdpr_recursive(user):
    field.anonymise(user=user)
```

`instance._log_gdpr_recursive(user=user, start=True)` would be called on enter and `instance._log_gdpr_recursive(user=user, start=False)` on exit?

Reply:
that would be fine.

Basically, since the whole thing is recursive, it's nice to know when a given object's recursion starts and stops. So a log line is saved for the start, then all the middle stuff, and then one at the end.

Later, we use this to reconstruct the history for anonymising a given object